index.js

```diff
@@ -174,7 +174,16 @@ InfluxDB.prototype.getDatabaseNames = function (callback) {
     if (err) {
       return callback(err, results)
     }
-    return callback(err, _.map(results[0].series[0].values, function (dbarray) { return dbarray[0] }))
+
+    if (_.isArray(results) && !_.isEmpty(results) && _.isArray(results[0].series) &&
+        !_.isEmpty(results[0].series) && _.isArray(results[0].series[0].values)) {
+
+      return callback(err, _.map(results[0].series[0].values, function (dbarray) {
+        if (_.isArray(dbarray) && !_.isEmpty(dbarray)) {
+          return dbarray[0]
+        }
+      }))
+    } else { return callback(new Error('bad response from server', results))}
```

Comment:
Hm, that works, though I'm kind of hesitant about going down the path of guarding against crafted responses. The simplest way to attack the app, which we could not feasibly guard against, would be for an attacker to create an enormous response and DoS the application. If an attacker has the capability to intercept calls you make to the database, you're already pretty screwed.